### PR TITLE
Refactor skycoin

### PR DESCRIPTION
**Pull request by therealssj**
Changes:
- create skycoin package
- run skycoin node cli using skycoin package
 

Does this change need to mentioned in CHANGELOG.md?
No

### CODE REVIEW

**Comment by gz-c** [comment]
conflicts

File: Gopkg.lock [hunk 1]
```diff
@@ -112,6 +112,6 @@
 [solve-meta]
   analyzer-name = "dep"
   analyzer-version = 1
-  inputs-digest = "454aacd6dc7c6819166ecc75a75483f4bcebc23ab61c4b2b32e694ee67020d52"
+  inputs-digest = "33d8d0c2675c0851febc914bfcadef2a117d2cc81db1cfea6ca00da527543f18"
```
Comment (gz-c) [inline Gopkg.lock]: why did this change?

File: cmd/skycoin/skycoin.go [hunk 2]
```diff
@@ -295,7 +101,7 @@ var devConfig = Config{
 
 	LaunchBrowser: false,
 	// Data directory holds app data -- defaults to ~/.skycoin
-	DataDirectory: filepath.Join(home, ".skycoin"),
+	DataDirectory: "~/.skycoin",
```
Comment (gz-c) [inline cmd/skycoin/skycoin.go]: I think we should use `$HOME` and manually substitute that

File: src/skycoin/config.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+// BuildConfig records build info
+type BuildConfig struct {
+	Version string // version number
+	Commit  string // git commit id
+	Branch  string // git branch name
+}
+
+// BlockchainConfig records genesis data
+type BlockchainConfig struct {
```
Comment (gz-c) [inline src/skycoin/config.go]: All of these can be merged into a single Config struct.  If we don't want to expose any as CLI parameters, we just won't register them with `flag`

File: Gopkg.lock (hunk 1 above)
Comment (therealssj) [reply Gopkg.lock]: oh thats becoz I had introduced some dependencies in an earlier commit but removed them now

File: cmd/skycoin/skycoin.go (hunk 2 above)
Comment (therealssj) [reply cmd/skycoin/skycoin.go]: yea right now I am substituting `~` with the home path.

File: src/skycoin/testdata/test.fiber.toml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	"172.104.85.6:6000",
+	"139.162.7.132:6000",
+]
+launch_browser = true
+peer_list_url = "https://downloads.skycoin.net/blockchain/peers.txt"
+
+[build]
+version = "0.23.1-rc2"
+commit = "0aab9bf7730827d6fd11beb0d02096b40cea1872"
+branch = "test-branch"
```
Comment (gz-c) [inline src/skycoin/testdata/test.fiber.toml]: make sure newline at end of files

File: fiber.toml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+#db_path = ""
+#db_read_only = false
+#arbitrating = false
+#log_to_file = false
+#version = false
+
+[build]
+version = "0.23.1-rc2"
+#commit = ""
+#branch = ""
```
Comment (gz-c) [inline fiber.toml]: make sure newline at end of files

File: Gopkg.lock (hunk 1 above)
Comment (gz-c) [reply Gopkg.lock]: still need to undo the deps change

**Comment by gz-c** [comment]
merge conflict

**Comment by therealssj** [comment]
@gz-c done

**Comment by gz-c** [comment]
Need to update vendored dependencies, according to vendorcheck

**Comment by gz-c** [comment]
build still failing for the same reason

**Pull request merged**